### DISCUSSION TITLE
Building menus in code

### DISCUSSION THREAD

**Discussion by ghost** [modern-forms/Modern.Forms#37] [Q&A] [answered]
What is the difference between MenuItem and MenuDropDown, and what are the use cases for each?

Is the hierarchy like...

Menu
----MenuItem
--------MenuItem
--------MenuItem
----MenuItem
--------MenuItem
--------MenuItem

...or...

Menu
----MenuDropDown
--------MenuItem
--------MenuItem
----MenuDropDown
--------MenuItem
--------MenuItem
?

When running ModernDecompile (on Linux Mint 21 Cinnamon) the Language menu flashes a small window where the submenu would be, but does not display. Contrast that with the Control Gallery sample app, which displays submenus OK for me. I cannot tell why there is a difference, but my project ([https://gitlab.com/sjsepan/ModernFormsApp1](https://gitlab.com/sjsepan/ModernFormsApp1)) does the flash and no display

**Comment by ghost** [comment]
Found a solution to the sub-menu flashing a window but not displaying:

in .csproj, point to clone of source instead of using a PackageReference (even though _dotnet add package Modern.Forms_ worked to add the reference):

`    
    <ProjectReference Include="..\Modern.Forms\src\Modern.Forms\Modern.Forms.csproj" />`
    
`    <!-- <PackageReference Include="Modern.Forms" Version="0.2.0" /> -->`


Note: this code is running against .Net 6.0

**Comment by jpobst** [comment, marked as answer]
I think the intent is to always use `MenuItem`.  I think originally it was `MenuDropDown`, but I was trying to simplify things by just making it always `MenuItem`.

The `0.2.0` package is over 2 years old, so it is definitely missing a lot of features and fixes.  I am thinking I'll try to put out `0.3.0` this month.  `ModernDecompile` runs on the `0.2.0` package, so it makes sense that it would display the flashing as well.
